test/functional/test_framework/test_framework.py

```diff
@@ -155,6 +159,15 @@ def main(self):
             exit_code = self.shutdown()
             sys.exit(exit_code)
 
+    def run_custom_test(self):
+        for method_name in self.options.test_methods:
+            method = getattr(self, method_name, None)
+            if not method:
+                raise AttributeError(f"No method with name {method_name}")
+            if not callable(method):
+                raise TypeError(f"{method_name} is not callable")
```

Comment:
Might as well find the searchable & callable methods initially and execute them while throwing the error for those that are filtered out? Otherwise few would be executed and the first incorrect one would throw error.

Reply:
Done, thanks